src/io/itk-dicom/dicom.cpp

```diff
@@ -112,6 +125,89 @@ void movefile(const std::string &src, const std::string &dst) {
   }
 }
 
+// doesn't actually do any length checks, or overflow checks, or anything
+// really.
+template <int N>
+double dotProduct(const std::vector<double> &vec1,
+                  const std::vector<double> &vec2) {
+  double result = 0;
+  for (int i = 0; i < N; i++) {
+    result += vec1.at(i) * vec2.at(i);
+  }
+  return result;
+}
+
+std::string encodeCosinesAsIDPart(const std::vector<double> &cosines) {
+  std::string concatenated;
+  for (auto it = cosines.begin(); it != cosines.end(); ++it) {
+    concatenated += std::to_string(*it);
+    if (it != cosines.end() - 1) {
+      concatenated += 'S';
+    }
+  }
+
+  replaceChars(concatenated, '-', 'N');
+  replaceChars(concatenated, '.', 'D');
+
+  return concatenated;
+}
+
+std::vector<double> ReadImageOrientationValue(const std::string &filename) {
+  gdcm::Reader reader;
+  reader.SetFileName(filename.c_str());
+  if (!reader.Read()) {
+    throw std::runtime_error("gdcm: failed to read file");
+  }
+  const gdcm::File &file = reader.GetFile();
+  // This helper method asserts that the vector has length 6.
+  return gdcm::ImageHelper::GetDirectionCosinesValue(file);
+}
+
+bool areCosinesAlmostEqual(std::vector<double> cosines1,
+                           std::vector<double> cosines2) {
+  for (int i = 0; i <= 1; i++) {
+    std::vector<double> vec1{cosines1.at(i), cosines1.at(i + 1),
+                             cosines1.at(i + 2)};
+    std::vector<double> vec2{cosines2.at(i), cosines2.at(i + 1),
+                             cosines2.at(i + 2)};
+    double dot = dotProduct<3>(vec1, vec2);
+    if (dot < (1 - 10e-5)) {
```

Comment:
should `10e-5` be an argument of `areCosinesAlmostEqual` with a default value?

Reply:
This should be a file-scoped static constant. I can also make it a default parameter, too.

Reply:
up to you. I think having it as a default parameter is a nice balance to avoid having to update if it that specific value isn't enough for some people/cases.